operator/operator_server.go

```diff
@@ -0,0 +1,118 @@
+package operator
+
+// since operator is a serverless component on its own, its associated server
+// is only used for hosting the /metrics, /livez and /readyz endpoints
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana-app-sdk/health"
+)
+
+type OperatorServer struct {
```

Comment:
Perhaps this should be named something like `ObservabilityServer`/`MetricsServer`/`HealthServer`, as it's not really exclusive to being an operator (and doesn't contain the operator endpoints for webhooks)? 

We may even want to put it in a different package to be shared by other things, but that's probably a future consideration for now.